pandas/io/parquet.py

```diff
@@ -194,14 +194,25 @@ def write(self, df, path, compression='snappy', **kwargs):
         # thriftpy/protocol/compact.py:339:
         # DeprecationWarning: tostring() is deprecated.
         # Use tobytes() instead.
-        path, _, _ = get_filepath_or_buffer(path)
+
+        if is_s3_url(path):
+            path, _, _ = get_filepath_or_buffer(path, mode='wb')
+            kwargs['open_with'] = lambda path, _: path
+        else:
+            path, _, _ = get_filepath_or_buffer(path)
+
         with catch_warnings(record=True):
             self.api.write(path, df,
                            compression=compression, **kwargs)
 
     def read(self, path, columns=None, **kwargs):
-        path, _, _ = get_filepath_or_buffer(path)
-        parquet_file = self.api.ParquetFile(path)
+        if is_s3_url(path):
+            s3, _, _ = get_filepath_or_buffer(path)
```

Comment:
is there a reason not to directly call the s3.get_filepath_or_buffer here?

Reply:
Not familiar with pandas code, i'm frankly not sure about existing design where `common.py#get_filepath_or_buffer` returns `s3file, or str, of buffer` but I'm holding back from making too many changes in my **first** PR... so I prefer to continue using what is implemented and working in for ex. `PyArrowImpl#read` where reading from `s3` works.